eventsourcing/src/test/java/org/axonframework/eventsourcing/DefaultSnapshotterSpanFactoryTest.java

```diff
@@ -0,0 +1,83 @@
+package org.axonframework.eventsourcing;
+
+import org.axonframework.tracing.IntermediateSpanFactoryTest;
+import org.axonframework.tracing.SpanFactory;
+import org.axonframework.tracing.TestSpanFactory;
+import org.junit.jupiter.api.Test;
+
+
+class DefaultSnapshotterSpanFactoryTest extends IntermediateSpanFactoryTest<DefaultSnapshotterSpanFactory.Builder, DefaultSnapshotterSpanFactory> {
```

Comment:
This `IntermediateSpanFactoryTest` class makes me think whether an `IntermediateSpanFactory` interface makes any sense or has any purpose for the multitude of intermediate span factories you will construct.

Reply:
I also thought about it being abstract and having an abstract builder for the spanFactory, but... I think it's over the top